include/seqan3/std/ranges

```diff
@@ -13,8 +13,45 @@
 
 #pragma once
 
-#if __cpp_lib_ranges // C++20 ranges available
+#if __has_include(<ranges>)
 #include <ranges>
+#endif // __has_include(<ranges>)
+
+#if defined(__cpp_lib_ranges) // C++20 ranges available
+#include <seqan3/std/iterator>
+
+#include <range/v3/range/concepts.hpp>
+
+namespace ranges
+{
+// std::ranges::views are valid range-v3 views
+template<class T>
+//!\cond
```

Comment:
This is not documented so you don't need the `cond`s here?

Reply:
Oh you are right, the think is doxygen does not evaluate the else branch in these constructions:

```c++
#if 1
...
#else
...
#endif
```

Reply:
So your point is to leave them anyway?

Reply:
I added a doxygen note.